### PR TITLE
Add TaskCountStatsMonitor to monitor task count stats

### PR DESCRIPTION
**Pull request by QiuMM**
Have tested in real cluster and now running in production.

### CODE REVIEW

File: server/src/main/java/org/apache/druid/server/metrics/TaskCountStatsMonitor.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+package org.apache.druid.server.metrics;
+
+import com.google.inject.Inject;
+import org.apache.druid.java.util.emitter.service.ServiceEmitter;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.apache.druid.java.util.metrics.AbstractMonitor;
+
+import java.util.Map;
+
+public class TaskCountStatsMonitor extends AbstractMonitor
```
Comment (jihoonson) [inline server/src/main/java/org/apache/druid/server/metrics/TaskCountStatsMonitor.java]: Would you please add a unit test? I think it should verify the values emitted from this monitor.

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskMaster.java [hunk 2]
```diff
@@ -41,14 +41,16 @@
...
 
+import java.util.Map;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.concurrent.locks.ReentrantLock;
 
 /**
  * Encapsulates the indexer leadership lifecycle.
  */
-public class TaskMaster
+public class TaskMaster implements TaskCountStatsProvider
```
[the remark this answers is not in the document]
Comment (QiuMM) [reply indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskMaster.java]: Hmm, `TaskCountStatsProvider` is bind to `TaskMaster`,  there is no need to let `TaskQueue` be `TaskCountStatsProvider`.

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java [hunk 3]
```diff
@@ -100,6 +103,11 @@
 
   private static final EmittingLogger log = new EmittingLogger(TaskQueue.class);
 
+  private final Map<String, AtomicLong> totalSuccessfulTaskCount = new ConcurrentHashMap<>();
+  private final Map<String, AtomicLong> totalFailedTaskCount = new ConcurrentHashMap<>();
+  private Map<String, Long> prevSuccessfulTaskCount = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (QiuMM) [reply indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java]: Done.

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java [hunk 4]
```diff
@@ -510,6 +518,14 @@ private void handleStatus(final TaskStatus status)
                     task,
                     status.getDuration()
                 );
+
+                if (status.isSuccess()) {
+                  totalSuccessfulTaskCount.computeIfAbsent(task.getDataSource(), k -> new AtomicLong());
+                  totalSuccessfulTaskCount.get(task.getDataSource()).incrementAndGet();
```
[the remark this answers is not in the document]
Comment (QiuMM) [reply indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java]: Good, done.

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java [hunk 5]
```diff
@@ -510,6 +518,14 @@ private void handleStatus(final TaskStatus status)
...
+                } else {
+                  totalFailedTaskCount.computeIfAbsent(task.getDataSource(), k -> new AtomicLong());
+                  totalFailedTaskCount.get(task.getDataSource()).incrementAndGet();
```
[the remark this answers is not in the document]
Comment (QiuMM) [reply indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java]: Good, done.

File: docs/content/operations/metrics.md [hunk 6]
```diff
@@ -163,6 +163,10 @@ Note: If the JVM does not support CPU time measurement for the current thread, i
 |`segment/added/bytes`|Size in bytes of new segments created.|dataSource, taskId, taskType, interval.|Varies.|
 |`segment/moved/bytes`|Size in bytes of segments moved/archived via the Move Task.|dataSource, taskId, taskType, interval.|Varies.|
 |`segment/nuked/bytes`|Size in bytes of segments deleted via the Kill Task.|dataSource, taskId, taskType, interval.|Varies.|
+|`task/success/count`|Number of successful tasks per emission period. This metric is only available if the TaskCountStatsMonitor module is included.|dataSource.|Varies.|
+|`task/failed/count`|Number of failed tasks per emission period. This metric is only available if the TaskCountStatsMonitor module is included.|dataSource.|Varies.|
+|`task/running/count`|Number of current running tasks. This metric is only available if the TaskCountStatsMonitor module is included.|dataSource.|Varies.|
+|`task/pending/count`|Number of current pending tasks. This metric is only available if the TaskCountStatsMonitor module is included.|dataSource.|Varies.|
```
[the remark this answers is not in the document]
Comment (QiuMM) [reply docs/content/operations/metrics.md]: Good, done.

File: indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java [hunk 7]
```diff
@@ -586,4 +602,36 @@ private void syncFromStorage()
     return rv;
   }
 
+  private Map<String, Long> getDeltaValues(Map<String, Long> total, Map<String, Long> prev)
+  {
+    return total.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue() - prev.getOrDefault(e.getKey(), 0L)));
```
[the remark this answers is not in the document]
Comment (QiuMM) [reply indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java]: Good, done.

File: server/src/main/java/org/apache/druid/server/metrics/TaskCountStatsMonitor.java (hunk 1 above)
Comment (QiuMM) [reply server/src/main/java/org/apache/druid/server/metrics/TaskCountStatsMonitor.java]: Done.

File: server/src/test/java/org/apache/druid/server/metrics/TaskCountStatsMonitorTest.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+  @Before
+  public void setUp()
+  {
+    statsProvider = new TaskCountStatsProvider()
+    {
+      @Override
+      public Map<String, Long> getSuccessfulTaskCount()
+      {
+        return null;
```
Comment (jihoonson) [inline server/src/test/java/org/apache/druid/server/metrics/TaskCountStatsMonitorTest.java]: Please fill all metrics.

**Pull request merged**